api/chat.py

```diff
@@ -110,7 +114,13 @@ def construct_prompt(query: str, history: List[Dict[str, str]], context: List[Bl
 
     question_prompt = f"In your answer, please cite any claims you make back to each source " \
                     f"using the format: [a], [b], etc. If you use multiple sources to make a claim " \
-                    f"cite all of them. For example: \"AGI is concerning [c, d, e].\"\n\nQ: " + query
+                    f"cite all of them. For example: \"AGI is concerning [c, d, e].\"" \
+                    "\n\nMaintain a tone that is vehemently pro-humanity, and treat the" \
+                    "issue with the gravity of a potentially existential threat. Do not" \
+                    "be afraid to take strong sides, nor feel every claim need be hedged" \
+                    "with its inverse. If something is important or true, state it" \
+                    "unequivocally. Stress that it’s imperative that we act soon." \
+                    "\n\n\nQ: " +query
```

Comment:
If you're going to update the prompt, can you please manually update the stampy-chat-test gsheet (https://docs.google.com/spreadsheets/d/12sX5uKvkZZRTIdFfUZKpYMVN-UCf5Op-bI8rU_9kzw8/edit#gid=385046677)? Add a tab with today's date and generated outputs for the benchmark questions so we can track how changes are affecting the results. I'm personally not convinced those wording are in the direction we necessarily want to head. But I'll defer if it produces desirable outcomes, as judged by Rob and distillation fellows. So it'll be even better if you can note examples where changing the prompt improved outputs.

Reply:
We have a general dev meeting tomorrow. If you're able to get a list of generated outputs to review before, that'd be nice. But don't stress over it if not.

> Maintain a tone that is vehemently pro-humanity, and treat the issue with the gravity of a potentially existential threat. Do not be afraid to take strong sides, nor feel every claim need be hedged with its inverse. If something is important or true, state it unequivocally. Stress that it’s imperative that we act soon.

In theory, the ChatGPT's RLHF should have induced pro-humanity. The prompt above seem likely to induce hallucinations which might reduce our credibility. So having output for reference would be helpful.

Reply:
Also nitpicky but for long strings, it might be easier to read & maintain using triple quotes, i.e.
```
question_prompt = """In your answer, 
please cite any claims you make back to each source using the format: 
[a], [b], etc. If you use multiple sources to make a claim 
cite all of them. For example: 'AGI is concerning [c, d, e].'

Maintain a tone that is vehemently pro-humanity and treat the
issue with the gravity of a potentially existential threat. Do not
be afraid to take strong sides, nor feel every claim need be hedged
with its inverse. If something is important or true, state it
unequivocally. Stress that it’s imperative that we act soon.

Q: """ + query
```

Reply:
Thanks for the note on the triple quote strings, I had forgotten python had those.

As for the prompt change, I really was just looking to test if the prompt eng tooling was useful. It seemed to have a positive effect on a few of the test questions so I left it in the PR. Will role it back now.

Reply:
Ohh I remember why I wasn't using triple quoted strings. It renders all newlines and indentation into the string itself - not viable.
<img width="824" alt="Screenshot 2023-07-09 at 2 43 25 PM" src="https://github.com/StampyAI/stampy-chat/assets/30442265/63e12bd6-6042-470b-8ecd-984e29c6053f">